dces-drc-integration/src/main/java/uk/gov/justice/laa/crime/dces/integration/service/ContributionService.java

```diff
@@ -92,107 +94,129 @@ public boolean processDailyFiles() {
             // Setup and make MAAT API "ATOMIC UPDATE" REST call below:
             LocalDateTime dateGenerated = LocalDateTime.now();
             String fileName = contributionsMapperUtils.generateFileName(dateGenerated);
             String xmlFile = contributionsMapperUtils.generateFileXML(successfulContributions.values().stream().toList(), fileName);
             String ackXml = contributionsMapperUtils.generateAckXML(fileName, dateGenerated.toLocalDate(), failedContributions.size(), successfulContributions.size());
-            List<String> successfulIdList = successfulContributions.keySet().stream().toList();
-            try {
-                contributionFileId = contributionUpdateRequest(xmlFile, successfulIdList, successfulIdList.size(), fileName, ackXml);
-                log.info("Created Concor contribution-file ID {} from {} Concor contribution IDs [{}]", contributionFileId, successfulIdList.size(), String.join(", ", successfulIdList));
-            } catch (MaatApiClientException | WebClientResponseException | HttpServerErrorException e) {
-                // We're rethrowing the exception, therefore avoid logging the stack trace to prevent logging the same trace multiple times.
-                String payload = "Failed to create Concor contribution-file. Investigation needed. State of files will be out of sync! [" + e.getClass().getName() + "(" + e.getMessage() + ")]";
-                // If failed, we want to handle this. As it will mean the whole process failed for current day.
-                // TODO: Need to figure how we're going to log a failed call to the ATOMIC UPDATE.
-                eventService.logConcor(null, UPDATED_IN_MAAT, batchId, null, INTERNAL_SERVER_ERROR, payload);
-                throw e;
-            }
+            List<BigInteger> successfulIdList = successfulContributions.keySet().stream().toList();
+            contributionFileId = executeConcorFileCreateCall(xmlFile, successfulIdList, successfulIdList.size(), fileName, ackXml);
         }
         logMaatUpdateEvents(successfulContributions, failedContributions);
-
         return contributionFileId;
     }
 
+
+    private CONTRIBUTIONS mapContributionXmlToObject(BigInteger concorContributionId, String xmlEntry, Map<BigInteger, String> failedContributions) {
+        CONTRIBUTIONS currentContribution;
+        try {
+            currentContribution = contributionsMapperUtils.mapLineXMLToObject(xmlEntry);
+        } catch (JAXBException e) {
+            failedContributions.put(concorContributionId, e.getClass().getName() + ": " + e.getMessage());
+            log.error("Failed to unmarshal contribution data XML, concorContributionId = {}", concorContributionId, e);
+            eventService.logConcor(concorContributionId, FETCHED_FROM_MAAT, batchId, null, INTERNAL_SERVER_ERROR, "Failed to unmarshal contribution data XML");
+            return null;
+        }
+
+        if (feature.outgoingAnonymized()) {
```

Comment:
Please add a log message here to say this is an effect of the feature flag, like done elsewhere.